tests/ut_textchannellistener/ut_textchannellistener.cpp

```diff
@@ -129,8 +129,11 @@ void Ut_TextChannelListener::cleanupTestCase()
         model.getGroups(IM_ACCOUNT_PATH);
         QVERIFY(waitSignal(modelReady, 5000));
 
-        while (model.rowCount() != 0)
+        while (model.rowCount() != 0) {
+            int waitForDelete = model.rowCount()-1;
             QVERIFY(model.deleteGroups(QList<int>() << model.group(model.index(0,0)).id()));
+            QTRY_COMPARE(model.rowCount(), waitForDelete);
```

Comment:
The deletion occurs synchronously, but the fact that groups are deleted is reported over D-Bus, and the model changes the row count in response to the IPC notification.